extension/test/background_test.ts

```diff
@@ -65,26 +87,76 @@ describe('background.ts', function () {
         { config: rcxMain.config }
       );
     });
+
+    it('should throw typeError if tab is undefined', async function () {
+      // }).to.throw(TypeError, 'sender.tab is always defined here.');
+    });
   });
-});
 
-async function sendMessageToBackground({
-  tabId = 0,
-  type,
-  responseCallback = () => {
-    // Do nothing by default.
-  },
-}: {
-  tabId?: number;
-  type: string;
-  responseCallback?: (response: unknown) => void;
-}): Promise<void> {
-  // In background.ts, a promise is passed to `addListener` so we can await it here.
-  // eslint-disable-next-line @typescript-eslint/await-thenable
-  await chrome.runtime.onMessage.addListener.yield(
-    { type: type },
-    { tab: { id: tabId } },
-    responseCallback
-  );
-  return;
-}
+  describe('xsearch', function () {
+    it('should return search results', async function () {
+      const searchStub = sinon.stub().returns(['to eat', 'verb']);
+      rcxMain.search = searchStub;
+      const request = {
+        type: 'xsearch',
+        text: 'た',
+        dictOption: '2',
```

Comment:
Not sure I follow. This is testing the `case: 'xsearch'` so if we don't pass it into the request, it won't be caught be the switch statement right?

Reply:
Sorry this comment is attached to line 103 so I meant `dictOption` when I said type. I think I forgot there was an actual key called `type` and just meant in general since this refers to the dictionary type.